controllers/tasks.py

```diff
@@ -0,0 +1,35 @@
+import webapp2
+from utils.decorators import user_required
+from utils import render_template, render_json
+from models.task import Task
+from models.project import Project	
+		
+class TaskHandler(webapp2.RequestHandler):
+	
+	@user_required
+	def get(self):
+		task_token = self.request.get('key')
+		task = Task.find_with_key(task_token)
+		params = {
+			'task': task
+		}
+		render_template(self, 'task.html', params)
+		
+	def post(self):
```

Comment:
Right idea, but wrong execution.  We're going to end up needing something that looks up the task based on some query parameter, which would look something like this:

```python
task_token = self.request.get('key')
task = Task.find_with_key(task_token)
```

then, we can set the new value for `completed` and save the model

```python
task.completed = True
task.put()
```

and finally finish the request

```python
render_json(self)
```